cedar-policy-validator/src/typecheck.rs

```diff
@@ -266,8 +266,15 @@ impl<'a> Typechecker<'a> {
     /// ensures that the policy expression has type boolean. If typechecking
     /// succeeds, then the method will return true, and no items will be
     /// added to the output list. Otherwise, the function returns false and the
-    /// output list is populated with any errors encountered while typechecking.
-    pub fn typecheck_policy(&self, t: &Template, type_errors: &mut HashSet<TypeError>) -> bool {
+    /// input `type_errors` list is populated with any errors encountered
+    /// while typechecking. In addition, warnings may be added to the `warnings`
+    /// list, although these will oot impact the boolean return value.
+    pub fn typecheck_policy(
+        &self,
+        t: &Template,
+        type_errors: &mut HashSet<TypeError>,
+        warnings: &mut HashSet<ValidationWarningKind>,
+    ) -> bool {
```

Comment:
this boolean return value used by anyone? or ignored by all callers?

should callers just check `type_errors.is_empty()` and `warnings.is_empty()` to be more explicit about whether they are checking for the absence of warnings or not?

Reply:
Return value is only used in tests -- I was on the fence about whether to remove it.

Reply:
Decided to remove in the latest commit.

Reply:
Scratch that -- turns out that the boolean value does matter sometimes. Will update comments accordingly.

Reply:
We can still remove it if we want, but it would be beyond the scope of this PR